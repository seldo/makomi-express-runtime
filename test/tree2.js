var mkRun = require('../index.js');

var layout = {
  source: "layouts/default",
  context: {
    "title": "I am a tree"
  },
  templates: {
    "body": {
      "source": "basic/tree2",
      "context": {},
      "templates": {
        "items": [
          {
            "source": "basic/treeitem",
            "context": {
              "name": "Branch1",
              "value": 1
            },
            "templates": {
              "children": {
                "source": "basic/tree2",
                "context": {},
                "templates": {
                  "items": [
                    {
                      "source": "basic/treeitem",
                      "context": {
                        "name": "Branch1-1",
                        "value": 11
                      },
                      "templates": {}
                    },
                    {
                      "source": "basic/treeitem",
                      "context": {
                        "name": "Branch1-2",
                        "value": 12
                      },
                      "templates": {
                        "children": {
                          "source": "basic/tree2",
                          "context": {},
                          "templates": {
                            "items": [
                              {
                                "source": "basic/treeitem",
                                "context": {
                                  "name": "Branch1-2-1",
                                  "value": 121
                                },
                                "templates": {}
                              },
                              {
                                "source": "basic/treeitem",
                                "context": {
                                  "name": "Branch1-2-2",
                                  "value": 122
                                },
                                "templates": {}
                              },
                              {
                                "source": "basic/treeitem",
                                "context": {
                                  "name": "Branch1-2-3",
                                  "value": 123
                                },
                                "templates": {}
                              }
                            ]
                          }
                        }
                      }
                    },
                    {
                      "source": "basic/treeitem",
                      "context": {
                        "name": "Branch1-3",
                        "value": 13
                      },
                      "templates": {}
                    }
                  ]
                }
              }
            }
          },
          {
            "source": "basic/treeitem",
            "context": {
              "name": "Branch2",
              "value": 2
            },
            "templates": {
              "children": {
                "source": "basic/tree2",
                "context": {},
                "templates": {
                  "items": [
                    {
                      "source": "basic/treeitem",
                      "context": {
                        "name": "Branch2-1",
                        "value": 21
                      }
                    },
                    {
                      "source": "basic/treeitem",
                      "context": {
                        "name": "Branch2-2",
                        "value": 22
                      }
                    },
                    {
                      "source": "basic/treeitem",
                      "context": {
                        "name": "Branch2-3",
                        "value": 23
                      }
                    },
                    {
                      "source": "basic/treeitem",
                      "context": {
                        "name": "Branch2-4",
                        "value": 24
                      }
                    }
                  ]
                }
              }
            }
          },
          {
            "source": "basic/treeitem",
            "context": {
              "name": "Branch3",
              "value": 3
            },
            "templates": {}
          }
        ]
      }
    }
  }
}

mkRun.compile(layout,function(renderedView) {
  console.log("View rendered:")
  console.log(renderedView)
},process.cwd()+'/test/views/')