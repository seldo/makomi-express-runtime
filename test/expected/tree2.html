<!DOCTYPE html>
<html>
<head>
  <title>I am a tree</title>
  <link rel='stylesheet' href='/stylesheets/core.css' />
</head>
<body>
<h1>I am a tree</h1>
<ul>
  <li>
  <b>Branch1</b>
  <ul>
  <li>
  <b>Branch1-1</b>

</li>
<li>
  <b>Branch1-2</b>
  <ul>
  <li>
  <b>Branch1-2-1</b>

</li>
<li>
  <b>Branch1-2-2</b>

</li>
<li>
  <b>Branch1-2-3</b>

</li>
</ul>
</li>
<li>
  <b>Branch1-3</b>

</li>
</ul>
</li>
<li>
  <b>Branch2</b>
  <ul>
  <li>
  <b>Branch2-1</b>

</li>
<li>
  <b>Branch2-2</b>

</li>
<li>
  <b>Branch2-3</b>

</li>
<li>
  <b>Branch2-4</b>

</li>
</ul>
</li>
<li>
  <b>Branch3</b>

</li>
</ul>
<footer>

</footer>
</body>
</html>